src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/DefaultEditorTemplates.cs

```diff
@@ -350,7 +350,7 @@ public static IHtmlContent EmailAddressInputTemplate(IHtmlHelper htmlHelper)
         public static IHtmlContent DateTimeInputTemplate(IHtmlHelper htmlHelper)
         {
             ApplyRfc3339DateFormattingIfNeeded(htmlHelper, "{0:yyyy-MM-ddTHH:mm:ss.fffK}");
-            return GenerateTextBox(htmlHelper, inputType: "datetime");
+            return GenerateTextBox(htmlHelper, inputType: "datetime-local");
```

Comment:
Not sure I understand this... I agree that perhaps if the user typed in `<input type="datetime" ... blah blah />` that we should preserve that. But I think in all other cases we should start rendering `type="datetime-local"`.